pkg/controller/gcp/compute/network_test.go

```diff
@@ -0,0 +1,658 @@
+			},
+			expectErr: true,
+		},
+		"ProviderRetrievalFailed": {
+			args: args{
+				cr: &v1alpha1.Network{},
+				c: &networkConnector{
+					kube: &test.MockClient{
+						MockGet: func(_ context.Context, key client.ObjectKey, obj runtime.Object) error {
+							return errors.New("i reject to work")
+						},
+					},
+				},
+			},
+			expectErr: true,
+		},
+		"CredFromSecretRetrievalFailed": {
+			args: args{
+				cr: &v1alpha1.Network{
+					Spec: v1alpha1.NetworkSpec{
+						ResourceSpec: corev1alpha1.ResourceSpec{
+							ProviderReference: &v1.ObjectReference{
+								Name:      testProviderName,
+								Namespace: testNamespace,
+							},
+						},
+						GCPNetworkSpec: v1alpha1.GCPNetworkSpec{
+							Name: testNetworkName,
+						},
+					},
+				},
+				c: &networkConnector{
+					kube: &test.MockClient{
+						MockGet: func(_ context.Context, key client.ObjectKey, obj runtime.Object) error {
+							switch o := obj.(type) {
+							case *gcpv1alpha1.Provider:
+								if diff := cmp.Diff(
+									meta.NamespacedNameOf(&v1.ObjectReference{Name: testProviderName, Namespace: testNamespace}),
+									key); diff != "" {
+									t.Errorf("r: -want, +got:\n%s", diff)
+								}
+								testProvider.DeepCopyInto(o)
+								return nil
+							case *v1.Secret:
+								return errors.New("i do not trust you with secrets")
+							}
+							return nil
+						},
+					},
+				},
+			},
+			expectErr: true,
+		},
+		"NewServiceFailed": {
+			args: args{
+				cr: &v1alpha1.Network{
+					Spec: v1alpha1.NetworkSpec{
+						ResourceSpec: corev1alpha1.ResourceSpec{
+							ProviderReference: &v1.ObjectReference{
+								Name:      testProviderName,
+								Namespace: testNamespace,
+							},
+						},
+						GCPNetworkSpec: v1alpha1.GCPNetworkSpec{
+							Name: testNetworkName,
+						},
+					},
+				},
+				c: &networkConnector{
+					kube: &test.MockClient{
+						MockGet: func(_ context.Context, key client.ObjectKey, obj runtime.Object) error {
+							switch o := obj.(type) {
+							case *gcpv1alpha1.Provider:
+								if diff := cmp.Diff(
+									meta.NamespacedNameOf(&v1.ObjectReference{Name: testProviderName, Namespace: testNamespace}),
+									key); diff != "" {
+									t.Errorf("r: -want, +got:\n%s", diff)
+								}
+								testProvider.DeepCopyInto(o)
+								return nil
+							case *v1.Secret:
+								testSecret.DeepCopyInto(o)
+								return nil
+							}
+							return nil
+						},
+					},
+				},
+				ns: func(_ context.Context, _ ...option.ClientOption) (*compute.Service, error) {
+					return nil, errors.New("i should fail")
+				},
+			},
+			expectErr: true,
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			tc.args.c.newServiceFn = tc.args.ns
+			_, err := tc.args.c.Connect(context.Background(), tc.args.cr)
+			if diff := cmp.Diff(tc.expectErr, err != nil); diff != "" {
+				t.Errorf("r: -want, +got:\n%s", diff)
+			}
+		})
+	}
+
+	// Type test
```

Comment:
This could be implemented as an entry in the test case slice; see examples in https://github.com/crossplaneio/stack-gcp/pull/4.